### ISSUE TITLE
Change autoload functionality

### ISSUE THREAD

**Issue by asmagill** [Hammerspoon/hammerspoon#1425]
Per a variety of discussions, I am entering this as a new issue so it can be tracked/addressed directly:

It has been proposed:

* Remove the ability to disable autoloading of modules.  This has always been an undocumented feature and untested in quite some time.  At present it is known that at least one one scenario of requiring `hs.window`, `hs.application`, and `hs.uielement` will fail to load properly without autoloading.

* move autoloading console messages into `hs.logger`

* examine other uses of `print` for status in core modules and move these into `hs.logger` as well.

Have I stated the intention correctly?

**Comment by cmsj** [comment]
seems like an accurate description to me, and I am +1 for all of these things :)

**Comment by asmagill** [comment]
@cmsj, should the print statements in `setup.lua` be handled through the default logger instance or should using the use of `hs.logger` only start when `hs._coresetup` is loaded?

**Comment by cmsj** [comment]
@asmagill hmm, I think it might well be a good idea to use hs.logger only after _coresetup is done - better to sacrifice a few lines of logging flexibility than to introduce dependencies to _coresetup imo.

**Comment by asmagill** [comment]
`hs._coresetup` already requires logger when it establishes the logging callback delegate currently used by LuaSkin.

This proposal just makes that use universal for any status/debug/detail output generated by Hammerspoon or it's modules so that the user can make a choice as to whether they want to see it or not.

Coupled with the LuaSkin logging changes proposed in #1426, you would be able to select this on a per module basis.

**Comment by cmsj** [comment]
Good point, +1 :)
